table-api/src/main/java/io/deephaven/api/agg/Distinct.java

```diff
@@ -0,0 +1,30 @@
+package io.deephaven.api.agg;
+
+import io.deephaven.api.JoinMatch;
+import org.immutables.value.Value.Default;
+import org.immutables.value.Value.Immutable;
+
+@Immutable
+public abstract class Distinct implements Aggregation {
+
+    public static Distinct of(JoinMatch match) {
+        return ImmutableDistinct.builder().match(match).build();
```

Comment:
The builder interface should probably be exposed to the end-user, or we need more `of` methods.

Reply:
The user can do `Distinct#withNulls` now.